R/datamapping-boxwhisker.R

```diff
@@ -0,0 +1,67 @@
+#' @title BoxWhiskerDataMapping
+#' @docType class
+#' @description  Class for Box Whisker Mapping
+#' @export
+BoxWhiskerDataMapping <- R6::R6Class(
+  "BoxWhiskerDataMapping",
+  inherit = XYDataMapping,
+  public = list(
+    quantiles = NULL,
+    outliers = NULL,
+    yMin = NULL,
+    lower = NULL,
+    middle = NULL,
+    upper = NULL,
+    yMax = NULL,
+
+    initialize = function(...,
+                              yMin = NULL,
+                              lower = NULL,
+                              middle = NULL,
+                              upper = NULL,
+                              yMax = NULL,
+                              quantiles = c(0.05, 0.25, 0.5, 0.75, 0.95),
+                              outliers = NULL) {
+      super$initialize(...)
+      self$quantiles <- sort(quantiles)
+      self$outliers <- outliers
+
+      self$yMin <- yMin %||% self$quantiles[1]
+      self$lower <- lower %||% self$quantiles[2]
+      self$middle <- middle %||% self$quantiles[3]
+      self$upper <- upper %||% self$quantiles[4]
+      self$yMax <- yMax %||% self$quantiles[5]
+
+      self$quantiles <- c(self$yMin, self$lower, self$middle, self$upper, self$yMax)
+    },
+
+    getQuantiles = function(data) {
+      # Redfine group and y while removing NA values
+      group <- ifnotnull(
+        self$x,
+        data[!is.na.data.frame(data[, self$y]), self$x],

```

Comment:
is this correct here? (I don't understand it really but 
for group you do
data[!is.na.data.frame(data[, self$y]), self$x],
and for y you do
data[!is.na.data.frame(data[, self$y]), self$y]

And it's not symetrical. Just want to make sure

Reply:
and it's is correct, this does not need to be evailuated twice

Reply:
Indeed, I will include the check self$x does not have NA as well.